### PR TITLE
[FFM-9918]: Proxy disconnect SDK stream when environment or APIkey is deleted

### PR DESCRIPTION
**Pull request by akiraqb**
```
[FFM-9918]: Proxy disconnect SDK stream when environment or APIkey is deleted
 ### What: 
When env is removed we should close all streams for given environment topic.
When APIKey is removed we should all streams for given APIKey
Check the auth token claims for sdk stream/polling request,  if env does not exist proxy should reject token
### Why:
Ensure proper handling of events between proxy and sdks.
 ### Testing:
Locally - with Java SDK to confirm behaviour.
```

[FFM-9918]: https://harness.atlassian.net/browse/FFM-9918?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: token/token_source.go [hunk 1]
```diff
@@ -43,8 +44,11 @@ func (a Source) GenerateToken(key string) (domain.Token, error) {
 		return domain.Token{}, fmt.Errorf("key %q not found", key)
 	}
 
+	//apiKey:= string(domain.NewAuthAPIKey())
```
Comment (jcox250) [inline token/token_source.go:47]: Should this be here?

File: config/pushpin/pushpin.conf [hunk 2]
```diff
@@ -19,7 +19,7 @@ stats_connection_ttl=120
 services=condure,zurl,pushpin-proxy,pushpin-handler
 
 # plain HTTP port that mongrel2 should listen on
-http_port=7000
+http_port=7999
```
Comment (jcox250) [inline config/pushpin/pushpin.conf:22]: Think you've accidentally committed this?

File: middleware/middleware.go [hunk 3]
```diff
@@ -72,6 +73,23 @@ func NewEchoAuthMiddleware(secret []byte, bypassAuth bool) echo.MiddlewareFunc {
...
+	}
+
+	// check if apikey is in cache
+	for _, k := range keys {
+		if key == k {
+			return true
+		}
+	}
+	return false
+}
```
Comment (jcox250) [inline middleware/middleware.go:91]: Maybe I'm being dumb here, but can we not just do this? 

```go
// Just check if the API key exists in the cache, if it doesn't then we must have removed it
key, err := authRepo.Get(ctx, claims.Key)
if err != nil {
   if errors.Is(err, domain.ErrCacheNotFound) {
        return false
   }
   return false
 }
 
 return true
```

Comment (akiraqb) [reply middleware/middleware.go:88]:
```
func isKeyInCache(repo keyLookUp, claims *domain.Claims) bool {
	key := claims.APIKey
	_, ok := repo.Get(context.Background(), domain.AuthAPIKey(key))
	if ok {
		return true
	}
	return false
}
```

**Review by jcox250 — APPROVED** [review]

**Pull request merged**